src/TTextEdit.cpp

```diff
@@ -602,7 +640,8 @@ int TTextEdit::drawGraphemeBackground(QPainter& painter, QVector<QColor>& fgColo
     }
     textRects.append(textRect);
     QColor bgColor;
-    if (Q_UNLIKELY(static_cast<bool>(attributes & TChar::Reverse) != charStyle.isSelected())) {
+    bool caretIsHere = mudlet::self()->isCaretModeEnabled() && mCaretLine == line && mCaretColumn == column;
+    if (Q_UNLIKELY(static_cast<bool>(attributes & TChar::Reverse) != (charStyle.isSelected() != caretIsHere))) {
```

Comment:
I'm not sure of this - originally it was to EXCLUSIVE-OR the reverse text mode with the reverse effect that selecting text does (so when combined the state is the same as if neither was applied) - what does the "CaratMod" mode do to how the text is/should be shown?

Reply:
This is to implement the caret mode’s cursor. It is displayed as a reversed-color character cell, similar to old terminals.

The change above reverses the color when the character cell has normal color, and uses normal color when it has reversed-color.

Reply:
So logically you have to handle all three conditions at the same time and apply the reverse effect when an ***odd*** number of them are true. Maybe the logic you have chosen will work, I'll have to think about it a bit more.